source/slang/slang-ir-lower-generics.cpp

```diff
@@ -124,6 +133,41 @@ namespace Slang
                 }
             }
             cloneInstDecorationsAndChildren(&cloneEnv, &sharedBuilderStorage, func, loweredFunc);
+            
+            // If the function returns a generic typed value, we need to turn it
+            // into an `out` parameter, since only the caller can allocate space
+            // for it.
+            auto oldFuncType = cast<IRFuncType>(func->getDataType());
+            if (isPolymorphicType(oldFuncType->getResultType()))
```

Comment:
One question I have as this pass starts to grow is: should some of these steps be handled as their own distinct IR passes? E.g., should we have a stand-alone IR pass does the transformation for functions with generic return types, and thus allow the generics lowering pass to ignore this detail (because it would already have been handled upstream)?